server/handlers.go

```diff
@@ -697,7 +695,7 @@ func (s *Server) handleToken(w http.ResponseWriter, r *http.Request) {
 	case grantTypePassword:
 		s.handlePasswordGrant(w, r, client)
 	default:
-		s.tokenErrHelper(w, errInvalidGrant, "", http.StatusBadRequest)
+		s.tokenErrHelper(w, errUnsupportedGrantType, "", http.StatusBadRequest)
```

Comment:
Some providers use the unsupported grant type error to indicate that this is a valid grant, but they do not allow the user to authorize with this grant.  However, I think for us, it is better to return "unsuppoerted_grant_type" here. Thanks!